src/pygama/raw/buffer_processor/buffer_processor.py

```diff
@@ -0,0 +1,345 @@
+        window_in_name in rb.lgdo.keys()
+    ):
+        # Now check if the window_in_name is a waveform table or not, if so we need to modify the t0s
+        if isinstance(rb.lgdo[window_in_name], lgdo.WaveformTable):
+            # modify the t0s
+            t0s = process_windowed_t0(
+                rb.lgdo[window_in_name].t0, rb.lgdo[window_in_name].dt, window_start_idx
+            )
+
+            # Window the waveform values
+            array_of_arrays = rb.lgdo[window_in_name].values
+            windowed_array_of_arrays = window_array_of_arrays(
+                array_of_arrays, window_start_idx, window_end_idx
+            )
+
+            # Write to waveform table and then to file
+            wf_table = lgdo.WaveformTable(
+                t0=t0s, dt=rb.lgdo[window_in_name].dt, values=windowed_array_of_arrays
+            )
+
+            # add this wf_table to the original table
+            rb.lgdo.add_field(window_out_name, wf_table, use_obj_size=True)
+
+        # otherwise, it's (hopefully) just an array of equal sized arrays
+        else:
+            array_of_arrays = rb.lgdo[window_in_name]
+            windowed_array_of_arrays = window_array_of_arrays(
+                array_of_arrays, window_start_idx, window_end_idx
+            )
+            rb.lgdo.add_field(
+                window_out_name, windowed_array_of_arrays, use_obj_size=True
+            )
+
+        return None
+
+    # otherwise, rb.lgdo is some other type and we only process it if the rb.out_name is the same as window_in_name
+    elif rb.out_name == window_in_name:
+        array_of_arrays = rb.lgdo
+        windowed_array_of_arrays = window_array_of_arrays(
+            array_of_arrays, window_start_idx, window_end_idx
+        )
+
+        rb.out_name = window_out_name
+        rb.lgdo = windowed_array_of_arrays
+
+        return None
+
+    else:
+        raise KeyError(f"{window_in_name} not a valid key for this RawBuffer")
+
+
+def window_array_of_arrays(
+    array_of_arrays: ArrayOfEqualSizedArrays, window_start_idx: int, window_end_idx: int
+) -> ArrayOfEqualSizedArrays:
+    r"""
+    Given an array of equal sized arrays, for each array it returns the view [window_start_idx:window_end_idx]
+    """
+    if isinstance(array_of_arrays, lgdo.ArrayOfEqualSizedArrays):
+        return array_of_arrays.nda[:, window_start_idx:window_end_idx]
+    else:
+        raise TypeError(
+            f"Do not know how to window an LGDO of type {type(array_of_arrays)}"
+        )
+
+
+def process_presum(
+    rb: RawBuffer, presum_obj: ArrayOfEqualSizedArrays, dsp_dict: dict, proc: str
+) -> lgdo.WaveformTable:
+    r"""
+    Finds the presum rate that was used to perform the DSP and writes this to a table.
+    The name of a valid :func:`pygama.lgdo.WaveformTable` that DSP was performed on is used to extract the original
+    ``dt`` so that they can be updated by the presum rate. Then the presummed table is added to the rb.lgdo
+
```

Comment:
Oh... maybe the dt array should be a second input to the presum DSP, and a second output should be added that is the output array's dt...